src/main/java/me/karakelley/http/ApplicationFactory.java

```diff
@@ -0,0 +1,19 @@
+package me.karakelley.http;
+
+import me.karakelley.http.filesystem.FileFinderCache;
+import me.karakelley.http.filesystem.PublicDirectory;
+import me.karakelley.http.filesystem.RealFileFinder;
+import me.karakelley.http.handlers.Application;
+import me.karakelley.http.handlers.Handler;
+
+public class ApplicationFactory {
+  private final String NO_DIRECTORY = "nodirectory";
```

Comment:
Yeah, I'd say giving special meaning to certain string values like this is almost always a bad idea. Even in languages where it's common (JavaScript) I still try to avoid this. What about just having two separate overloads? `create(publicDirectory)` and `create()`, or just two separate methods? `createWithPublicDirectory`/`createWithoutPublicDirectory`.

Reply:
But at some point I need to check if a directory is present within the args, right?